Tanktweaks

SunshineDistillery opened the pull request:
Disassembly recipes to turn extinguishers and oxygen tanks into little metal tanks for crafting.

Coolthulhu commented:
Gases aren't supported yet.
Deconstruction recipe is OK, but it should waste the oxygen, not produce an itemized oxygen.

Coolthulhu commented on data/json/items/containers.json:
```diff
@@ -524,13 +524,13 @@
-        "volume" : 5,
-        "bashing" : 2,
+        "weight" : 800,
+        "volume" : 8,
```
The volume of container is also its capacity. In case of extinguishers and oxygen tank, it should be lower, because there are extra parts there.
If you still want it to get the boost, there are few vehicle parts that use little metal tanks. Those should get a proportional boost to capacity too.

Coolthulhu commented on data/json/items/tools.json:
```diff
@@ -405,10 +405,10 @@
     "price": 5000,
-    "material": "aluminum",
-    "weight": 9100,
+    "material": "steel",
```
Changing to steel and (much) lower weight?
It's supposed to be a "large emergency fire extinguisher".

SunshineDistillery commented:
The dis-assembly recipe doesn't recover oxygen from the medical tank. 

The other type of oxygen, the ammo, I just put in a container. Right now, even though the description says it's in a tank, it's actually some kind of solid steel material that can be divided in stacks. As a stop gap, until the game has gasses and air-tight/pressurized containers, I could make the ammo-oxygen a liquid, so it would at least require a container and could be spilled.  That's still not good, but it's better than the metal dust it is now. 

Weights for items are all over the place. Fire extinguishers were super heavy compared to steel tanks. I guess I'll google for realistic weights. Instead of reducing volume, I might increase weight a bit and then have the deconstruction recipe give some scrap and plastic chunks to represent all the parts and valves you wrench off.

Coolthulhu commented:
> As a stop gap, until the game has gasses and air-tight/pressurized containers, I could make the ammo-oxygen a liquid

No, that's still not good enough. Putting pure, pressurized oxygen in a waterskin should not work.
For now the only viable solution is keeping oxygen as charges of the oxygen tank. There is no good way to make itemized oxygen work with current system.
Look at hydrogen - it also only exists in metal tanks right now.

> Instead of reducing volume, I might increase weight a bit and then have the deconstruction recipe give some scrap and plastic chunks to represent all the parts and valves you wrench off. 

Makes sense.
Extinguisher could have a rubber hose in the disassembly recipe (instead of, or in addition to chunks) - many (most?) of those come with hoses.

SunshineDistillery commented:
I changed Oxy ammo back until there is a better way to handle it.  

Extinguisher volume was changed to match metal tank sizes. The small version is now pretty small, like the kind you might have on a kitchen countertop or in a car; the large one is the size of a big industrial extinguisher. I added their volumes in the descriptions since 'large' and 'small' are a little vague.

The deconstruction recipes now give some scrap and the large one gives a rubber hose.

kevingranade commented:
The only reasonable stopgap would be to lose the contents of a gas tank if
it's used for crafting, just a thought.
